packages/dds/map/src/test/mocha/map.order.spec.ts

```diff
@@ -0,0 +1,280 @@
+			map.set("2", 1);
+			map.set("1", 2);
+			map.set("3", 3);
+			map.set("1", 4);
+			map.set("4", 5);
+			map.set("2", 6);
+
+			assert.deepEqual(Array.from(map.keys()), ["2", "1", "3", "4"]);
+			assert.deepEqual(Array.from(map.values()), [6, 4, 3, 5]);
+		});
+
+		it("delete", () => {
+			map.delete("3");
+			map.set("2", 1);
+			map.set("1", 2);
+			map.set("3", 3);
+
+			map.delete("1");
+			assert.deepEqual(Array.from(map.keys()), ["2", "3"]);
+
+			map.set("1", 4);
+			assert.deepEqual(Array.from(map.keys()), ["2", "3", "1"]);
+			assert.deepEqual(Array.from(map.values()), [1, 3, 4]);
+		});
+
+		it("clear", () => {
+			map.set("2", 1);
+			map.set("1", 2);
+			map.set("3", 3);
+
+			map.clear();
+			assert.deepEqual(Array.from(map.keys()), []);
+			assert.deepEqual(Array.from(map.values()), []);
+
+			map.set("3", 1);
+			map.set("2", 2);
+			map.delete("3");
+			map.set("1", 3);
+
+			assert.deepEqual(Array.from(map.keys()), ["2", "1"]);
+		});
+
+		it.skip("serialize/load", async () => {
+			map.set("2", 1);
+			map.set("1", 2);
+			map.set("4", 5);
+
+			const containerRuntimeFactory = new MockContainerRuntimeFactory();
+			const dataStoreRuntime = new MockFluidDataStoreRuntime();
+			const containerRuntime =
+				containerRuntimeFactory.createContainerRuntime(dataStoreRuntime);
+			const services = MockSharedObjectServices.createFromSummary(
+				map.getAttachSummary().summary,
+			);
+			services.deltaConnection = dataStoreRuntime.createDeltaConnection();
+
+			const loadedMap = new SharedMap("loadedMap", dataStoreRuntime, MapFactory.Attributes);
+			await loadedMap.load(services);
+
+			assert.deepEqual(Array.from(loadedMap.keys()), ["2", "1", "4"]);
+		});
+	});
+
+	describe("Connected state", () => {
+		let containerRuntimeFactory: MockContainerRuntimeFactory;
+		let map1: SharedMap;
+		let map2: SharedMap;
+
+		beforeEach(async () => {
+			containerRuntimeFactory = new MockContainerRuntimeFactory();
+			map1 = createConnectedMap("map1", containerRuntimeFactory);
+			map2 = createConnectedMap("map2", containerRuntimeFactory);
+		});
+
+		it("Remote messages have no conflict with local pending ops", () => {
+			map1.set("1", 1);
+			map1.set("2", 2);
+			map2.set("3", 1);
+			map2.set("4", 2);
+			map1.delete("1");
+
+			containerRuntimeFactory.processSomeMessages(2);
+			assert.deepEqual(Array.from(map1.keys()), ["2"]);
+			assert.deepEqual(Array.from(map2.keys()), ["1", "2", "3", "4"]);
+
+			containerRuntimeFactory.processSomeMessages(2);
+			assert.deepEqual(Array.from(map1.keys()), ["2", "3", "4"]);
+			assert.deepEqual(Array.from(map2.keys()), ["1", "2", "3", "4"]);
+
+			containerRuntimeFactory.processAllMessages();
+			assert.deepEqual(Array.from(map1.keys()), ["2", "3", "4"]);
+			assert.deepEqual(Array.from(map2.keys()), ["2", "3", "4"]);
+		});
+
+		it("Remote set conflicts with local pending set", () => {
+			map1.set("1", 1);
+			map1.set("2", 2);
+			map2.set("3", 3);
+			map2.set("1", 2);
```

Comment:
This test seems like it'd be useful to have assertions before processing any messages as well (more explicitly demonstrate that from map2's perspective, the key "1" 'jumps' to before the key "3" when map1's op is acked)